src/fu-device-list.c

```diff
@@ -168,22 +168,24 @@ fu_device_list_find_by_guid (FuDeviceList *self, const gchar *guid)
 }
 
 static FuDeviceItem *
-fu_device_list_find_by_platform_id (FuDeviceList *self, const gchar *platform_id)
+fu_device_list_find_by_connection (FuDeviceList *self, const gchar *platform_id, const gchar *logical_id)
 {
-	if (platform_id == NULL)
+	if (platform_id == NULL && logical_id == NULL)
```

Comment:
Not sure -- I think logical_id should be allowed to be NULL. I think matching physical_id=physical_id and NULL=NULL would be exactly what's expected.